src/plugins/discover/common/utils/sorting/get_sort_for_search_source.ts

```diff
@@ -31,16 +40,34 @@ export function getSortForSearchSource(
       return [{ _score: defaultDirection } as EsQuerySortValue];
     }
   }
+
   const { timeFieldName } = dataView;
-  return getSort(sort, dataView).map((sortPair: Record<string, string>) => {
-    if (dataView.isTimeNanosBased() && timeFieldName && sortPair[timeFieldName]) {
-      return {
-        [timeFieldName]: {
-          order: sortPair[timeFieldName],
-          numeric_type: 'date_nanos',
-        },
-      } as EsQuerySortValue;
+  const sortPairs = getSort(sort, dataView);
+
+  const sortForSearchSource = sortPairs.map((sortPair: Record<string, string>) => {
+    if (timeFieldName && sortPair[timeFieldName]) {
+      return getESQuerySortForTimeField({
+        sortDir: sortPair[timeFieldName] as SortDirection,
+        timeFieldName,
+        isTimeNanosBased: dataView.isTimeNanosBased(),
+      });
     }
     return sortPair as EsQuerySortValue;
   });
+
+  // Do we need to have a tie breaker like this? We have it for Surrounding Documents page but not here yet.
```

Comment:
@weltenwort Good point, thanks! Would you suggest to reuse `context:tieBreakerFields` setting or have a fixed tie breaker like `_shard_doc`?

Reply:
If I understand the docs correctly `_shard_doc` is only available within PIT contexts, although it would be a great option. But maybe I'm mistaken there. Otherwise it seems reasonable to me to re-use the existing tiebreaker setting.